### PR TITLE
AST-623 - fix: Null border size showing solid border for headers on frontend

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
- Customizer data - https://share.bsf.io/WnuxqlGD
- If empty (null) value is set from customizer still headers applying border style CSS
- Slack conversation - https://brainstormforce.slack.com/archives/C5QL6NXEK/p1624455426078500

### Screenshots
- https://share.getcloudapp.com/7KuoAQDw

### Types of changes
- Bug fix 

### How has this been tested?
- Import the above customizer data
- In this, the above header border is empty but it still shows the border for above header in frontend
- Ideal case should be - If & only if we apply border size then only it show border styles 

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by abhishek-bsf — DISMISSED** [review]

File: inc/builder/type/header/above-header/dynamic-css/dynamic.css.php [hunk 1]
```diff
@@ -82,8 +77,18 @@ function astra_above_header_row_setting( $dynamic_css, $dynamic_css_filtered = '
 			'line-height' => astra_get_css_value( $hba_header_height_desktop, 'px' ),
 		),
 	);
+
+	// Apply border only when it has positive value.
+	if ( $hba_header_divider ) {
```
Comment (premanshup) [inline inc/builder/type/header/above-header/dynamic-css/dynamic.css.php:82]: @imnavanath This condition will work even if the value is string like `inherit`, can you please add some check here that the value is greater than or equal to 0 only then we add this CSS?

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/builder/type/header/primary-header/dynamic-css/dynamic.css.php [hunk 2]
```diff
@@ -105,28 +105,32 @@ function astra_primary_header_breakpoint_style( $dynamic_css, $dynamic_css_filte
...
-			'border-bottom-width' => astra_get_css_value( $header_separator, 'px' ),
-			'border-bottom-color' => esc_attr( $header_separator_color ),
-			'border-bottom-style' => 'solid',
-		),
-	);
+	$header_separator = astra_get_option( 'hb-header-main-sep' );
 
-	$parse_css .= astra_parse_css( $meta_style );
+	// Apply border only when it has positive value.
+	if ( $header_separator ) {
```
Comment (premanshup) [inline inc/builder/type/header/primary-header/dynamic-css/dynamic.css.php:111]: @imnavanath Same here

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/builder/type/header/above-header/dynamic-css/dynamic.css.php (hunk 1 above)
Comment (imnavanath) [reply inc/builder/type/header/above-header/dynamic-css/dynamic.css.php:82]: Added `absint` here and in following comment's code @premanshup, so it is working correctly now, Please review [this commit](https://github.com/brainstormforce/astra/pull/2915/commits/9eabecdc55288c02cd3c17c7e965e16006a43bbf)

File: inc/builder/type/header/primary-header/dynamic-css/dynamic.css.php (hunk 2 above)
Comment (imnavanath) [reply inc/builder/type/header/primary-header/dynamic-css/dynamic.css.php:111]: Added `absint` here [review this commit](https://github.com/brainstormforce/astra/pull/2915/commits/9eabecdc55288c02cd3c17c7e965e16006a43bbf)

**Review by premanshup — APPROVED** [review]

**Review by shubhw12 — DISMISSED** [review]

**Pull request merged**